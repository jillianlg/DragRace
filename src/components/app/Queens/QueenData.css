.div{
  display: flex;
  width: auto;
}

section{
  border: solid pink 1px;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
  font-size: 14px;
  width: 50vw;

  p{
    padding: 0px 10px 0px 10px;
  }
}