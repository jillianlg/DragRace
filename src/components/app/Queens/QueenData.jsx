import React from 'react';
import PropTypes from 'prop-types';

const QueenData = ({ image_url, name, seasons:[season], quote }) => (
  <div data-testid="details">
    <p>{name}</p>
    <img style={{ height: '200px' }} src={image_url} alt={name} />
    <p>{season.id}</p>
    <p>{quote}</p>
  </div>
);

QueenData.propTypes = {
  name: PropTypes.string.isRequired,
  image_url: PropTypes.string.isRequired,
  seasons: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired,
};

export default QueenData;
