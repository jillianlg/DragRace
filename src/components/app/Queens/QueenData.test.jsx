import React from 'react';
import { render, cleanup } from '@testing-library/react';
import QueenData from './QueenData';



describe('QueenData component', () => {
  afterEach(() => cleanup());
  it('renders Queen data by id', () => {
    const { asFragment } = render(
      <QueenData
        name="Bianca Del Rio"
        image_url="BDR.com"
        seasons="6"
        quote="Not today Satan."
      />);
    expect(asFragment()).toMatchSnapshot();
  });
});
