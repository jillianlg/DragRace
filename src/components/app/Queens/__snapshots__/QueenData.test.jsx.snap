// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueenData component renders Queen data by id 1`] = `
<DocumentFragment>
  <div
    data-testid="details"
  >
    <p>
      Bianca Del Rio
    </p>
    <img
      alt="Bianca Del Rio"
      src="BDR.com"
      style="height: 200px;"
    />
    <p />
    <p>
      Not today Satan.
    </p>
  </div>
</DocumentFragment>
`;
